lib/internal/streams/async_iterator.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+const kLastResolve = Symbol('lastResolve');
+const kLastReject = Symbol('lastReject');
+const kError = Symbol('error');
+const kEnded = Symbol('ended');
+const kLastPromise = Symbol('lastPromise');
+const kHandlePromise = Symbol('handlePromise');
+
+class AsyncIteratorRecord {
```

Comment:
Use `const AsyncIteratorRecord = class AsyncIteratorRecord { ... }` to avoid assignment guards at the use sites.

Reply:
Can you elaborate on this, @bmeurer? This might be something we need to do elsewhere in our code...

Reply:
I mentioned this before in http://benediktmeurer.de/2017/06/29/javascript-optimization-patterns-part2 In V8 we track information about `C` for each `new C` use site. As long as you always pass exactly the same `C`, we just emit a guard in TurboFan code that checks whether the value you pass for `C` is exactly the same value we recorded. This check is redundant (and get's eliminated) if `C` is a known constant (in TurboFan). `class` bindings are mutable, whereas `const` bindings are immutable, so using `const` here makes the check on each use site redundant and avoids it in the optimized machine code.

Reply:
Thanks for the detailed explanation!

Reply:
That being said, you're right, this might be something we need to do elsewhere in the code as well. Maybe it deserves an eslint rule?